### PR TITLE
[#3322] Fix all dotnet compiler warnings in BotBuilder-Samples

### PR DESCRIPTION
**Pull request by erquirogasw**
Addresses # 3322
#minor

## Description
This PR fixes all dotnet compiler warnings in BotBuilder-Samples.

## Specific Changes
Fixed warnings:

- Obsolete related warnings
- Async related warnings
- .NET Framework version related warnings
- Razor language version downgrade warnings

## Testing

Inspection.
image

StateManagement.
image

Scaleout.
image

TeamsAuth.
image

TeamsMessagingExtensionsAction.
image

QnAMaker-All-Features.
image

Skills-SSO-CloudAdapter.
image

### CODE REVIEW

File: samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs [hunk 1]
```diff
@@ -10,13 +10,23 @@
...
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
         // This method gets called by the runtime. Use this method to add services to the container.
-        public void ConfigureServices(IServiceCollection services)
+        public void ConfigureServices(IServiceCollection services, BotComponent botcomponent)
```
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs]:
```suggestion
        public void ConfigureServices(IServiceCollection services, BotComponent botComponent)
```

File: samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs [hunk 2]
```diff
@@ -38,7 +48,9 @@ public void ConfigureServices(IServiceCollection services)
             // Create the bot as a transient. In this case the ASP Controller is expecting an IBot.
             services.AddTransient<IBot, CustomQABot<RootDialog>>();
 
-            ComponentRegistration.Add(new DialogsComponentRegistration());
+            //ComponentRegistration.Add(new DialogsComponentRegistration());
```
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs]: Remove unused code

File: samples/csharp_dotnetcore/49.qnamaker-all-features/Startup.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public Startup(IConfiguration configuration)
         public IConfiguration Configuration { get; }
 
         // This method gets called by the runtime. Use this method to add services to the container.
-        public void ConfigureServices(IServiceCollection services)
+        public void ConfigureServices(IServiceCollection services, BotComponent botcomponent)
```
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/49.qnamaker-all-features/Startup.cs]: same here

File: samples/csharp_dotnetcore/49.qnamaker-all-features/Startup.cs [hunk 4]
```diff
@@ -53,7 +53,9 @@ public void ConfigureServices(IServiceCollection services)
             // Create the bot as a transient. In this case the ASP Controller is expecting an IBot.
             services.AddTransient<IBot, QnABot<RootDialog>>();
 
-            ComponentRegistration.Add(new DialogsComponentRegistration());
+            //ComponentRegistration.Add(new DialogsComponentRegistration());
```
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/49.qnamaker-all-features/Startup.cs]: same here

File: samples/csharp_dotnetcore/46.teams-auth/TeamsAuth.csproj [hunk 5]
```diff
@@ -7,7 +7,6 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.1" />
-    <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.1.2" PrivateAssets="All" />
```
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/46.teams-auth/TeamsAuth.csproj:10]: Are we sure removing this package the bot keeps working as expected, the login, logout, everything?, don't know but maybe this package someone forgot to remove it?

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs]: What if we update the methods to be synchronous? Like this:
```CSharp
protected IQnAMakerClient GetQnAMakerClient()
{
    return _services?.QnAMakerService;
}
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: samples/csharp_dotnetcore/46.teams-auth/TeamsAuth.csproj (hunk 5 above)
Comment (ceciliaavila) [reply samples/csharp_dotnetcore/46.teams-auth/TeamsAuth.csproj:10]: I think we can remove it, but we'll check it when we test the sample.

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs]: This warning may be fixed by returning `await Task.FromResult(new MessagingExtensionActionResponse());` (We should try it)

File: samples/csharp_dotnetcore/82.skills-sso-cloudadapter/RootBot/TokenExchangeSkillHandler.cs [hunk 8]
```diff
@@ -39,6 +41,7 @@ public class TokenExchangeSkillHandler : CloudSkillHandler
             BotFrameworkAuthentication auth,
             SkillsConfiguration skillsConfiguration,
             IConfiguration configuration,
+            ConfigurationBotFrameworkAuthentication configurationBotFrameworkAuthentication,
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/82.skills-sso-cloudadapter/RootBot/TokenExchangeSkillHandler.cs]: The constructor is missing the property setting. 
And probably you should add this new parameter when you instantiates the class.
```CSharp
_configurationBotFrameworkAuthentication = configurationBotFrameworkAuthentication ?? throw new ArgumentNullException(nameof(configurationBotFrameworkAuthentication));
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs (hunk 6 above)
Comment (erquirogasw) [reply samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs]: Don't we need the DialogContext?

File: samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs (hunk 2 above)
Comment (erquirogasw) [reply samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs]: Done. Thanks!

File: samples/csharp_dotnetcore/49.qnamaker-all-features/Startup.cs (hunk 4 above)
Comment (erquirogasw) [reply samples/csharp_dotnetcore/49.qnamaker-all-features/Startup.cs]: Done

File: samples/csharp_dotnetcore/46.teams-auth/TeamsAuth.csproj (hunk 5 above)
Comment (erquirogasw) [reply samples/csharp_dotnetcore/46.teams-auth/TeamsAuth.csproj:10]: The sample was built successfully. Anyway, I'll test all paths locally.

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs (hunk 7 above)
Comment (erquirogasw) [reply samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs]: Done. Thanks!

File: samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs (hunk 6 above)
Comment (ceciliaavila) [reply samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs]: It wasn't used. Anyway, I think it'd be better to keep the override of the method and add the `Task.FromResult()` as I suggested for the teams sample.

File: samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs [hunk 9]
```diff
@@ -37,8 +47,8 @@ public void ConfigureServices(IServiceCollection services)
 
             // Create the bot as a transient. In this case the ASP Controller is expecting an IBot.
             services.AddTransient<IBot, CustomQABot<RootDialog>>();
-
-            ComponentRegistration.Add(new DialogsComponentRegistration());
+
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/48.customQABot-all-features/Startup.cs]: These extra spaces can be removed.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Just a small change.

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Pull request closed without merging**